experiments/react-native-uifabric/src/components/PersonaCoin/PersonaCoin.helpers.ts

```diff
@@ -0,0 +1,102 @@
+  },
+  busy: {
+    uri:
+      'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAMAAADXqc3KAAAANlBMVEUAAADpABvpABvpABvpABvpABvpABvpABvpABvpABvpABvpABvpABvpABvpABvpABvpABvpABvsgIdoAAAAEXRSTlMA8eLUEQj6s2sheXrlyrJqHBqb3nMAAAB/SURBVCjPhZFRDsQgCEQFEdS223L/yy7JummtGN/nvASUCSvoFGSIclIXpwragC3deYn6IJZ/fqF24NXmHPri+E2rOlCDQTAKIBNZHbKJjyfEBHoCTbAn2ATMRPRHzZfPn+t/cH4SIw3rY2p17H2+l0VRRtr8ag3KgswomcKCL1NmGBp9lqsZAAAAAElFTkSuQmCC'
+  }
+};
+
+// TODO: Need icons for the OOF statues
+const presenceOOFIconCache = presenceIconCache;
+
+export function getPresenceIconSource(presence: PersonaPresence, isOutOfOffice: boolean): ImageURISource {
+  return isOutOfOffice ? presenceOOFIconCache[presence] : presenceIconCache[presence];
+}
+
+export type PersonaSizeConfig = {
+  physicalSize: number;
+  iconSize: number;
+  initialsSize: number;
+};
+
+const sizeTable: { [P in PersonaSize]: PersonaSizeConfig } = {
+  size8: { physicalSize: 8, iconSize: 0, initialsSize: 4 },
+  size24: { physicalSize: 24, iconSize: 8, initialsSize: 10 },
+  size32: { physicalSize: 32, iconSize: 8, initialsSize: 12 },
+  size40: { physicalSize: 40, iconSize: 12, initialsSize: 14 },
+  size48: { physicalSize: 48, iconSize: 12, initialsSize: 16 },
+  size56: { physicalSize: 56, iconSize: 16, initialsSize: 18 },
+  size72: { physicalSize: 72, iconSize: 20, initialsSize: 20 },
+  size100: { physicalSize: 100, iconSize: 28, initialsSize: 36 },
+  size120: { physicalSize: 120, iconSize: 32, initialsSize: 40 }
+};
+
+export function getSizeConfig(size: PersonaSize): PersonaSizeConfig {
+  return sizeTable[size];
+}
+
+const colorTable: { [P in PersonaCoinColor]: string } = {
+  lightBlue: '#4F6BED',
+  blue: '#0078D4',
+  darkBlue: '#004E8C',
+  teal: '#038387',
+  green: '#498205',
+  darkGreen: '#0B6A0B',
+  lightPink: '#C239B3',
+  pink: '#E3008C',
+  magenta: '#881798',
+  purple: '#5C2E91',
+  orange: '#CA5010',
+  lightRed: '#D13438',
+  darkRed: '#A4262C',
+  violet: '#8764B8',
+  gold: '#986F0B',
+  burgundy: '#750B1C',
+  warmGray: '#7A7574',
+  cyan: '#005B70',
+  rust: '#8E562E',
+  coolGray: '#69797E'
+};
+
+export function convertCoinColor(coinColor: PersonaCoinColor): string {
+  return colorTable[coinColor];
+}
+
+export function calculateEffectiveSizes(tokens: IPersonaCoinTokens): PersonaSizeConfig {
+  const { size, coinSize, iconSize, initialsSize } = tokens;
+
+  if (size) {
+    return sizeTable[size];

```

Comment:
It is probably best to apply the size value, then patch with coinSize, iconSize, and initialsSize as overrides.  So this could then become something like:

    const sizeEntry = size && sizeTable[size] || sizeTable['size40'];
    return {
      physicalSize: coinSize || sizeEntry.coinSize,
      etc...
    };

Reply:
To follow up a bit I think this is a good general practice for our APIs, for instance a fontVariant property might set fontFace, fontSize, and fontWeight.  It should be valid for a user to specify something like: fontVariant="header", fontWeight=500 to basically override one property specifically.